### PR TITLE
Add documentation for new component BP5758D LED driver.

### PR DESCRIPTION
**Pull request by Cossid**
## Description:
Add documentation for new component BP5758D LED driver

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4021

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by Cossid** [comment]
Note to reviewer: Lint failure is because of a reference link to another document also in (a separate) PR.

File: components/output/bp5758d.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    should be set to true when using this driver for safest operation.
+
+See Also
+--------
+
+- :doc:`/components/output/index`
+- :doc:`/components/output/esp8266_pwm`
+- :doc:`/components/output/sm2135`
+- :doc:`/components/output/sm2335`
+- :doc:`/components/output/bp1658cj`
```
Comment (jesserockz) [inline components/output/bp5758d.rst]: The site wont actually build with this error. You can add this link in the second PR as I will merge this one first
```suggestion
```

File: components/output/bp5758d.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        channel: 1 # Valid values 1-5 to match OUT1-OUT5
+        current: 10 # In mA, valid values 0-90, default 10
+      - platform: bp5758d
+        id: output_coldwhite
+        channel: 5 # Valid values 1-5 to match OUT1-OUT5
+        current: 10 # In mA, valid values 0-90, default 10
+      - platform: bp5758d
+        id: output_warmwhite
+        channel: 4 # Valid values 1-5 to match OUT1-OUT5
+        current: 10 # In mA, valid values 0-90, default 10
```
Comment (jesserockz) [inline components/output/bp5758d.rst]: This example should only have the hub code, then below is a single float output. 
If you want to have a full example for a specific bulb, it should go down the bottom above `see also`
```suggestion
```

File: components/output/bp5758d.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
```
Comment (jesserockz) [inline components/output/bp5758d.rst]:
```suggestion
    # Example configuration entry
    output:
      - platform: bp5758d
        id: output_red
        channel: 2 # Valid values 1-5 to match OUT1-OUT5
        current: 10 # In mA, valid values 0-90, default 10
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Comment by jesserockz** [comment]
Haha, I just realised you have another PR for the `sm2335` as well. Maybe remove all crosslinking between the pages in the PRs and make one more PR that links them once they are all merged. (Small PRs with no dependencies are easy to merge :smile:)

**Comment by Cossid** [comment]
Yeah, I have 3 LED drivers in queue (BP5758D, BP1658CJ, and SM2335/SM2235).  I'll unlink them from each other in docs shortly.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**